fuels-contract/src/contract.rs

```diff
@@ -38,8 +39,11 @@ pub struct CallResponse<D> {
 }
 
 impl Contract {
-    pub fn new(compiled_contract: CompiledContract) -> Self {
-        Self { compiled_contract }
+    pub fn new(compiled_contract: CompiledContract, wallet: LocalWallet) -> Self {
```

Comment:
Resolved after internal discussion (on Slack)? Summary: we're following the workflow from `ethers-rs` such that you bind the signer (`wallet`) to the contract instance in order to avoid passing a signer to every contract call (including contract deployment).

Reply:
Will document the birdview in the merge commit